CMakeLists.txt

```diff
@@ -2,6 +2,10 @@ project(MONGO_CXX_DRIVER LANGUAGES CXX)
 
 cmake_minimum_required(VERSION 3.1.0 FATAL_ERROR)
 
+set (CMAKE_SKIP_BUILD_RPATH false)
+set (CMAKE_BUILD_WITH_INSTALL_RPATH false)
+set (CMAKE_INSTALL_RPATH_USE_LINK_PATH true)
+
```

Comment:
Lets read this and make sure that this is really what we want: https://cmake.org/Wiki/CMake_RPATH_handling